hazelcast/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -396,7 +396,7 @@ private void runTasklet(TaskletTracker t) {
                     CancellationException ex = (CancellationException) e;
                     t.executionTracker.exception(ex);
                 } else {
-                    logger.warning("Exception in " + t.tasklet, e);
+                    logger.fine("Exception in " + t.tasklet, e);
```

Comment:
If there is an exception, e.g. from `result = t.call();` do we get the whole stacktrace reported on the master? We have some issues with incomplete stacktraces, I am not sure if this is the case or it's something else.

If we have the full stacktrace, then I am fine with this change.

Alternatively we could have a special case for the `QueryResultSizeExceededException` similar to `CancellationException` above.

Reply:
To me it feels a little wrong to add a special case for a specific job/implementation detail to here, because `CancellationException` is generic, a user can cancel any of the jobs, but 'QueryResultSizeExceededException' doesnt feel so generic, so if we were to add a special case I d suggest we add interfaces to exceptions that could occur in a job that denotes how they are logged, might be a little more generic. WDYT about this?

About the stack trace, I only changed the logging level, idk if that effects how the stacktrace is reported, but I will see and inform here how the stacktrace is reported.

Reply:
Currently, there is 1 special case, we would add 2nd one. Let's wait for a 3rd before we try to create an abstraction for this.

> changed the logging level
yes, and after this change, we won't see the log on the member. This is desired for `QueryResultSizeExceededException` but for any other exception, this is interesting, so unless we are sure (I am not) that the exception is correctly propagated and logged on master I would keep it also on the member at warning level.

Reply:
Before, the default logging level was WARN and it was reduced to INFO for cancellation. Now the default is FINE, but is _increased_ for cancellation. That doesn't make sense.

But Frantisek's point is right. Users can have a plethora of causes for job failures. I think there are cases when the client doesn't receive the full stack trace, even though I was unable find such scenario. And I didn't look into non-java clients. So I think it makes sense to log on member with INFO level by default, and decrease the level to FINE for the two cases we know are normal: for cancellation and SqlLimitExc.

Reply:
@frant-hartm WDYT about the approach Viliam suggested?

Reply:
Fine with me.

Reply:
So I tried and on the Master node an Error message that occurred during the execution of a job is printed with a stacktrace even on INFO level. Thus, I believe it is okey to use INFO as the default level and FINE for cancellation and SqlLimitException for now.